src/evict.c

```diff
@@ -546,8 +546,8 @@ int performEvictions(void) {
         goto update_metrics;
     }
 
-    if (server.maxmemory_policy == MAXMEMORY_NO_EVICTION) {
-        result = EVICT_FAIL; /* We need to free memory, but policy forbids. */
+    if (server.maxmemory_policy == MAXMEMORY_NO_EVICTION || server.pseudo_replica) {
```

Comment:
it's better to place it in `isSafeToPerformEvictions` together with the `server.primary_host` check.

Reply:
If place it in `isSafeToPerformEvictions `, the server will ignore maxmemory. I think it's better to return an OOM error to stop the syncing process.